lib/gssapi/ntlm/init_sec_context.c

```diff
@@ -381,6 +381,7 @@ _gss_ntlm_init_sec_context
 		if (RAND_bytes(nonce, sizeof(nonce)) != 1) {
 		    _gss_ntlm_delete_sec_context(minor_status,
 						 context_handle, NULL);
+		    heim_ntlm_free_type2(&type2);
```

Comment:
I think these are all the leak points for type2 but this function needs refactoring to make it reviewable.